parser/eam.go

```diff
@@ -0,0 +1,134 @@
+package parser
+
+import (
+	"bytes"
+	"encoding/hex"
+	"github.com/filecoin-project/go-state-types/builtin/v10/eam"
+	filTypes "github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+	"github.com/ipfs/go-cid"
+	"github.com/zondax/fil-parser/types"
+)
+
+func (p *Parser) parseEam(txType string, msg *filTypes.Message, msgRct *filTypes.MessageReceipt, msgCid cid.Cid, ethLogs []types.EthLog) (map[string]interface{}, error) {
+	metadata := make(map[string]interface{})
+	switch txType {
+	case MethodConstructor:
+	case MethodCreate:
+		return p.parseCreate(msg.Params, msgRct.Return, msgCid)
+	case MethodCreate2:
+		return p.parseCreate2(msg.Params, msgRct.Return, msgCid)
+	case MethodCreateExternal:
+		return p.parseCreateExternal(msg, msgRct, msgCid, ethLogs)
+	}
```

Comment:
You can use the same parser for the three methods, the return params are the same:
this is from lotus code:
```go
type CreateReturn Return
type Create2Return Return
type CreateExternalReturn Return
```